### PR TITLE
fixing issue throughout

### PR DESCRIPTION
**Pull request by bmullan91**
Enables the IS NULL when the query parameter is an empty string.

### CODE REVIEW

File: lib/postgresql-store.js [hunk 1]
```diff
@@ -401,8 +401,14 @@ module.exports = function (opts) {
     var wherestr = ''
 
     if (!_.isEmpty(w) && w.params.length > 0) {
-      w.params.forEach(function (param) {
-        params.push('"' + escapeStr(camelToSnakeCase(param)) + '"=$' + (++cnt))
+      w.params.forEach(function (param, i) {
+        if(w.values[i] === null || w.values[i] === '') {
```
Comment (nherment) [inline lib/postgresql-store.js]: @bmullan91 empty string should not create a WHERE IS NULL criteria.

empty string is a valid value and we reduce the functionality if we do that.

File: lib/postgresql-store.js [hunk 2]
```diff
@@ -437,8 +443,14 @@ module.exports = function (opts) {
     var wherestr = ''
 
     if (!_.isEmpty(w) && w.params.length > 0) {
-      w.params.forEach(function (param) {
-        params.push('"'+escapeStr(camelToSnakeCase(param)) + '"=$' + (++cnt))
+      w.params.forEach(function (param, i) {
+        if(w.values[i] === null || w.values[i] === '') {
```
Comment (nherment) [inline lib/postgresql-store.js]: same here, ``'' !== null``

File: lib/postgresql-store.js [hunk 3]
```diff
@@ -485,7 +497,7 @@ module.exports = function (opts) {
 
     if (!_.isEmpty(w) && w.params.length > 0) {
       w.params.forEach(function (param, i) {
-        if(w.values[i] === null) {
+        if(w.values[i] === null || w.values[i] === '') {
```
Comment (nherment) [inline lib/postgresql-store.js]: same here, ``'' !== null``

**Comment by nherment** [comment]
@bmullan91 it's excellent to add the NULL capability to 'delete' and 'distinct' but we can't assume that a query criteria with an empty string value is equivalent to a null value. The two are different both in JS and in the DB.

**Comment by bmullan91** [comment]
Totally - I'll do those in a separate PR. This one can die.

**Pull request closed without merging**